datamap/mv/core/src/main/scala/org/apache/carbondata/mv/datamap/MVHelper.scala

```diff
@@ -172,6 +172,17 @@ object MVHelper {
     if (catalog.isMVWithSameQueryPresent(logicalPlan)) {
       throw new UnsupportedOperationException("MV with same query present")
     }
+
+    var expressionValid = true
+    modularPlan.transformExpressions {
+      case coal@Coalesce(_) =>
```

Comment:
what about `sum(coalesce(id, 0))`, this should  be supported while creating MV otherwise we cannot create MV in this scenario, Please add testcase for this as well and support it.

Reply:
done